File: packages/shared/types/newsletter.ts
export interface SubscribeRequest {
  email: string
}

export interface SubscribeResponse {
  message: string
}

export interface SubscriberResponse {
  id: string
  email: string
  status: 'en_attente' | 'actif' | 'desinscrit'
  confirmed_at: string | null
  unsubscribed_at: string | null
  created_at: string
}

export interface PaginatedSubscribers {
  items: SubscriberResponse[]
  total: number
  page: number
  per_page: number
}

File: apps/backend/app/schemas/newsletter.py
from datetime import datetime

from pydantic import BaseModel, EmailStr


class SubscribeRequest(BaseModel):
    email: EmailStr


class SubscribeResponse(BaseModel):
    message: str


class ConfirmResponse(BaseModel):
    message: str


class SubscriberResponse(BaseModel):
    id: str
    email: str
    status: str
    confirmed_at: datetime | None = None
    unsubscribed_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class PaginatedSubscribers(BaseModel):
    items: list[SubscriberResponse]
    total: int
    page: int
    per_page: int
